system/temperature.c

```diff
@@ -21,6 +21,141 @@
 
 #include "temperature.h"
 
+static uint16_t pci_read_pid(int bus, int dev, int func)
```

Comment:
With such trivial implementations, I think that `pci_read_pid()` and `pci_read_vid()` ought to be static inline functions in system/pci.h .

Reply:
This patch is basically a copy/paste of the related k8temp function without taking into consideration the use done by Memtest86+ (which is very different to k8temp) and its already-implemented functions. This can be extremely simplified without any feature loss. I'll try to do it later today.